mobilecoind/src/payments.rs

```diff
@@ -361,10 +358,14 @@ impl<T: BlockchainConnection + UserTxConnection + 'static, FPR: FogPubkeyResolve
 
         let num_blocks_in_ledger = self.ledger_db.num_blocks()?;
 
-        // Figure out the block_version and fee (involves network round-trips to
-        // consensus, unless fee arg is non-zero)
+        // Figure out the block version, fee and minimum fee map.
```

Comment:
:thought_balloon: I'm not saying it should be done now, or possibly ever in this case, but there are 3 instances of
```rust
        // Figure out the block version, fee and minimum fee map.
        let last_block_info = get_majority_block_info(last_block_infos)
            .ok_or_else(|| Error::TxBuild("No block info available".into()))?;

        let (fee, block_version) =
            self.get_network_fee_and_block_version(token_id, opt_fee, &last_block_info)?;

        let fee_map = FeeMap::try_from(last_block_info.minimum_fees)?;

        // Make sure we understand this block version
        let block_version =
            BlockVersion::try_from(block_version).map_err(|err| Error::TxBuild(err.to_string()))?;
```
It feels like that could be one method that provides the the `fee`, `fee_map`, and `block_version`, or errors

Reply:
`get_fee_info_and_block_version`?

Reply:
yah, it's local/private so seems good enough...